modules/sso.php

```diff
@@ -461,13 +462,109 @@ private function should_hide_login_form() {
 		return apply_filters( 'jetpack_remove_login_form', get_option( 'jetpack_sso_remove_login_form', false ) );
 	}
 
+	/**
+	 * Outputs the Jetpack SSO button and description as well as the toggle link
+	 * for switching between Jetpack SSO and default login.
+	 */
 	function login_form() {
-		$classes = '';
+		$site_name = get_bloginfo( 'name' );
+		if ( ! $site_name ) {
+			$site_name = get_bloginfo( 'url' );
+		}
 
-		if ( $this->should_hide_login_form() ) {
-			$classes .= ' forced-sso';
+		$display_name = ! empty( $_COOKIE[ 'jetpack_sso_wpcom_name' . COOKIEHASH ] )
+			? $_COOKIE[ 'jetpack_sso_wpcom_name' . COOKIEHASH ]
+			: false;
+		$gravatar = ! empty( $_COOKIE[ 'jetpack_sso_wpcom_gravatar' . COOKIEHASH ] )
+			? $_COOKIE[ 'jetpack_sso_wpcom_gravatar' . COOKIEHASH ]
+			: false;
+
+		?>
+		<div id="jetpack-sso-wrap">
+			<?php if ( $display_name && $gravatar ) : ?>
+				<div id="jetpack-sso-wrap__user">
+					<img width="72" height="72" src="<?php echo esc_html( $gravatar ); ?>" />
+
+					<h2>
+						<?php
+							_e(
```

Comment:
@ebinnion this should be:
```php
echo wp_kses(
    sprintf( __( 'Log in as <span>%s</span>', 'jetpack' ), esc_html( $display_name ) ) )
    array( 'span' => true )
);
```
the string to translate must always be static. In this case, a dynamic string, the result of `sprintf` was being passed to `_e`. I also added `wp_kses` allowing only the `span` since we can't escape all the string. The `esc_html` on the `$display_name` is perfect.

There's a similar case a little below with `You can now save time spent logging in by connecting your WordPress.com account to %s`